spec/unit/system_file_spec.rb

```diff
@@ -64,6 +65,16 @@
       type: "remote_dir"
     )
   }
+  let(:single_quote_file) {
+    Machinery::SystemFile.new(
+      name: "/bla/single-quote-file'-foo/"
```

Comment:
This looks like a directory but is called file so I would remove the backslash at the end.